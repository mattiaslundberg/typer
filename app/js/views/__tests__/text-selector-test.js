import React from 'react'
import TextSelector from '../text-selector'
import sd from 'skin-deep'

describe("TextSelectorTest", () => {
  const render = (cb=jest.fn()) => {
    return sd.shallowRender(
      <TextSelector onSelect={cb} options={[{"fulltext": "some full text", "name": "One"}]}/>
    )
  }

  it('Render selectable items', () => {
    let tree = render()
    let instance = tree.getMountedInstance()

    expect(tree.getRenderOutput()).toMatchSnapshot()
  })

  it('Returns to select view after cancel', () => {
    let tree = render()
    let instance = tree.getMountedInstance()
    instance.onAddClick()
    instance.onNewCancel()

    expect(tree.getRenderOutput().props.children).toMatchSnapshot()
  })
})
