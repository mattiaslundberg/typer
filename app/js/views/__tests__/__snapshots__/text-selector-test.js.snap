exports[`TextSelectorTest Render selectable items 1`] = `
<function Selection(props) {/* istanbul ignore next */_classCallCheck(this, Selection);var _this = _possibleConstructorReturn(this, (Selection.__proto__ || Object.getPrototypeOf(Selection)).call(this,
    props));

    /* istanbul ignore next */_this.onChange = /* istanbul ignore next */_this.onChange.bind( /* istanbul ignore next */_this);
    /* istanbul ignore next */_this.prepareText = /* istanbul ignore next */_this.prepareText.bind( /* istanbul ignore next */_this); /* istanbul ignore next */return _this;
  }
  onAddClick={[Function]}
  onSelect={[Function]}
  options={
    Array [
      Object {
        "fulltext": "some full text",
        "name": "One",
      },
    ]
  } />
`;

exports[`TextSelectorTest Returns to select view after cancel 1`] = `undefined`;
